pkg/localstore/localstore.go

```diff
@@ -344,7 +347,57 @@ func New(path string, baseKey []byte, ss storage.StateStorer, o *Options, logger
 			}
 		}
 		sharkyBase = &dirFS{basedir: sharkyBasePath}
+
+		// check whether file already existed, and if so, initiate recovery process
+		if isDirtyShutdown(path) {
+			//recovery
+			locOrErr, err := recovery(db)
+			if err != nil {
+				return nil, err
+			}
+
+			recoverySharky, err := sharky.NewRecovery(sharkyBasePath, sharkyNoOfShards, sharkyPerShardLimit, swarm.ChunkWithSpanSize)
+			if err != nil {
+				return nil, err
+			}
+
+			for l := range locOrErr {
+				if l.err != nil {
+					db.logger.Warning("error reading sharky location", l.err)
```

Comment:
We should return here I think. We can get error here on db.ctx closing, in which case we should return.